fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java

```diff
@@ -3451,9 +3451,9 @@ private void handleLoanRepaymentInFull(final LocalDate transactionDate, final Lo
             }
         }
         if (isAllChargesPaid) {
+            this.closedOnDate = transactionDate;
             loanLifecycleStateMachine.transition(LoanEvent.REPAID_IN_FULL, this);
 
-            this.closedOnDate = transactionDate;
             this.actualMaturityDate = transactionDate;
```

Comment:
Please move the actualMaturityDate as well...

Reply:
Done!